src/transactions/ManageOfferOpFrame.cpp

```diff
@@ -168,6 +168,9 @@ ManageOfferOpFrame::doApply(Application& app, LedgerDelta& delta,
         }
 
         mSellSheepOffer->storeDelete(tempDelta, db);
+        bool res = mSourceAccount->addNumEntries(-1, ledgerManager);
+        assert(res);
+        mSourceAccount->storeChange(tempDelta, db);
```

Comment:
`maxAmountOfSheepCanSell` is not computed properly anymore as code would end up in the `else` branch https://github.com/jonjove/stellar-core/blob/cf0063534781b5ecdb829fb3243f7b18ae8b1134/src/transactions/ManageOfferOpFrame.cpp#L227

I am surprised that tests didn't catch this.

Reply:
Added a test to catch this.